pkg/cmd/variable/shared/shared.go

```diff
@@ -0,0 +1,317 @@
+package shared
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/cli/cli/v2/api"
+	"github.com/cli/cli/v2/internal/config"
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/internal/text"
+	"github.com/cli/cli/v2/pkg/iostreams"
+)
+
+type Visibility string
+
+const (
+	All      = "all"
+	Private  = "private"
+	Selected = "selected"
+)
+
+type App string
+
+const (
+	Actions = "actions"
+	Unknown = "unknown"
+)
+
+func (app App) String() string {
+	return string(app)
+}
+
+func (app App) Title() string {
+	return text.Title(app.String())
+}
+
+type VariableEntity string
+
+const (
+	Repository   = "repository"
+	Organization = "organization"
+	Environment  = "environment"
+)
+
+type VariablePayload struct {
+	Name         string  `json:"name,omitempty"`
+	Value        string  `json:"value,omitempty"`
+	Visibility   string  `json:"visibility,omitempty"`
+	Repositories []int64 `json:"selected_repository_ids,omitempty"`
+}
+
+type repoNamesResult struct {
+	Ids []int64
+	Err error
+}
+
+type PostPatchOptions struct {
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+	Config     func() (config.Config, error)
+	BaseRepo   func() (ghrepo.Interface, error)
+
+	VariableName    string
+	OrgName         string
+	EnvName         string
+	Body            string
+	Visibility      string
+	RepositoryNames []string
+	Prompter        iprompter
+}
+
+type ListOptions struct {
```

Comment:
This is not being used in the PR. Let's remove all code from this PR that does not actually contribute to the `set` command.

Reply:
List Variables is required to determine if the operation is  patch on 409.